src/library_browser.js

```diff
@@ -1367,6 +1367,25 @@ var LibraryBrowser = {
     exit(status);
   },
 
+  emscripten_get_window_title__proxy: 'sync',
+  emscripten_get_window_title__sig: 'iv',
+  emscripten_get_window_title: function() {
+    return allocate(intArrayFromString(document.title), 'i8', ALLOC_STACK);
```

Comment:
Allocating this on the stack is too risky, I think - if the pointer is used after the calling function unwinds the stack, it might be overwritten. How about having mallocing the space, and saving a single such allocation? (reusing it on the next call). And the docs can mention that the string is valid until the next call to the function.

Reply:
Sounds good, is there an example of this I could use as a reference? It also looks like `_malloc` might be a better option here.

Reply:
I think this is a somewhat similar example: https://github.com/emscripten-core/emscripten/blob/7ba0b851d68544dee49c69231a260e26653bc233/src/library.js#L3808 , of a singleton allocation in a function.